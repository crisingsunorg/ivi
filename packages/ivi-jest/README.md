# Jest utils for ivi library
