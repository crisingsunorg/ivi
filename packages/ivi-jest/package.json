{
  "private": true,
  "name": "ivi-jest",
  "version": "0.0.1",
  "main": "dist/main/index.js",
  "module": "dist/module/index.js",
  "sideEffects": false,
  "files": [
    "dist",
    "src",
    "!dist/**/*.tsbuildinfo",
    "!src/**/__tests__",
    "README.md",
    "LICENSE"
  ],
  "scripts": {
    "prepublishOnly": "npm run clean && npm run dist",
    "clean": "rm -rf ./dist",
    "dist": "npm run dist:main && npm run dist:module",
    "dist:main": "../../node_modules/.bin/tsc -p ./tsconfig.build.main.json",
    "dist:module": "../../node_modules/.bin/tsc -p ./tsconfig.build.module.json"
  },
  "dependencies": {
    "jsdom": "^14.0.0"
  },
  "peerDependencies": {},
  "devDependencies": {},
  "description": "Jest utils for ivi library",
  "license": "MIT",
  "author": {
    "name": "Boris Kaul",
    "email": "localvoid@gmail.com",
    "url": "https://github.com/localvoid"
  },
  "keywords": [
    "ivi"
  ],
  "homepage": "https://github.com/localvoid/ivi",
  "repository": {
    "type": "git",
    "url": "https://github.com/localvoid/ivi"
  },
  "bugs": {
    "url": "https://github.com/localvoid/ivi/issues"
  },
  "github": "https://github.com/localvoid/ivi"
}
