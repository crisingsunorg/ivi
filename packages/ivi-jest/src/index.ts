export { useMockFn, useSpyOn } from "./mock";
export { useResetModules, useModule } from "./module";
export { useDOMElement, useDOMOpsCounters } from "./dom";
