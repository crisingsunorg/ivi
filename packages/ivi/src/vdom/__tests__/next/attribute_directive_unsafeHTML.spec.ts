import { useResetModules, useModule, useDOMElement, useSpyOn } from "ivi-jest";

useResetModules();
const c = useDOMElement();
const ivi = useModule<typeof import("ivi")>("ivi");
const h = useModule<typeof import("ivi-html")>("ivi-html");
const t = useModule<typeof import("ivi-test")>("ivi-test");
const innerHTML = useSpyOn(() => Element.prototype, "innerHTML", "set");
const _ = void 0;

describe("attribute directive unsafeHTML", () => {
  test("mount", () => {
    const { domNode } = t.render(h.div(_, { unsafeHTML: ivi.UNSAFE_HTML("<span>1</span>") }), c());
    expect(innerHTML().mock.calls.length).toBe(1);
    expect(domNode).toMatchSnapshot();
  });

  test("mount empty string", () => {
    const { domNode } = t.render(h.div(_, { unsafeHTML: ivi.UNSAFE_HTML("") }), c());
    expect(innerHTML().mock.calls.length).toBe(0);
    expect(domNode).toMatchSnapshot();
  });

  test("update", () => {
    t.render<any>(h.div(_, { unsafeHTML: ivi.UNSAFE_HTML("<span>1</span>") }), c());
    const { domNode } = t.render(h.div(_, { unsafeHTML: ivi.UNSAFE_HTML("<span>2</span>") }), c());
    expect(innerHTML().mock.calls.length).toBe(2);
    expect(domNode).toMatchSnapshot();
  });

  test("update with the same value", () => {
    const a = t.render<HTMLSpanElement>(h.div(_, { unsafeHTML: ivi.UNSAFE_HTML("<span>1</span>") }), c());
    const b = t.render<HTMLSpanElement>(h.div(_, { unsafeHTML: ivi.UNSAFE_HTML("<span>1</span>") }), c());
    expect(innerHTML().mock.calls.length).toBe(1);
    expect(a.domNode!.firstChild).toBe(b.domNode!.firstChild);
  });

  test("update with empty string should not assign innerHTML", () => {
    t.render(h.div(_, { unsafeHTML: ivi.UNSAFE_HTML("") }), c());
    const { domNode } = t.render(h.div(_, { unsafeHTML: ivi.UNSAFE_HTML("") }), c());
    expect(innerHTML().mock.calls.length).toBe(0);
    expect(domNode).toMatchSnapshot();
  });
});
