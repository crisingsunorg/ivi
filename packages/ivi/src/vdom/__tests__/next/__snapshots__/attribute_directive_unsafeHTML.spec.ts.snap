// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`attribute directive unsafeHTML mount 1`] = `
<div>
  <span>
    1
  </span>
</div>
`;

exports[`attribute directive unsafeHTML mount empty string 1`] = `<div />`;

exports[`attribute directive unsafeHTML update 1`] = `
<div>
  <span>
    2
  </span>
</div>
`;

exports[`attribute directive unsafeHTML update with empty string should not assign innerHTML 1`] = `<div />`;
